admin/category_product_listing.php

```diff
@@ -25,1230 +25,1656 @@
+                    $products_query_raw .= $extra_from;
+
+                    $products_query_raw .= " LEFT JOIN " . TABLE_PRODUCTS_DESCRIPTION . " pd ON (pd.products_id = p.products_id)";
+                    $products_query_raw .= $extra_joins;
+
+                    $where = " WHERE pd.language_id = " . (int)$_SESSION['languages_id'];
+                    $where .= $extra_ands;
+
+                    if ($search_result && $action !== 'edit_category') {
+                        $keyword_search_fields = [
+                            'pd.products_name',
+                            'p.products_model',
+                            'pd.products_description',
+                            'p.products_id',
+                        ];
+                        $where .= zen_build_keyword_where_clause($keyword_search_fields, trim($keywords));
+                    } else {
+                        $products_query_raw .= " LEFT JOIN " . TABLE_PRODUCTS_TO_CATEGORIES . " p2c ON (p2c.products_id = p.products_id) ";
+                        $where .= " AND p2c.categories_id=" . (int)$current_category_id;
+                    }
+
+                    $products_query_raw .= $where . $order_by;
+
+                    // Split Page
+
+                    // reset page when page is unknown
+                    if ((empty($_GET['page']) || $_GET['page'] === 1) && !empty($_GET['pID'])) {
+                        $check_page = $db->Execute($products_query_raw);
+                        if ($check_page->RecordCount() > $max_results) {
+                            $check_count = 0;
+                            foreach ($check_page as $item) {
+                                if ((int)$item['products_id'] === $_GET['pID']) {
+                                    break;
+                                }
+                                $check_count++;
+                            }
+                            $_GET['page'] = round((($check_count / $max_results) + (fmod_round($check_count, $max_results) != 0 ? .5 : 0)));
+                            $page = $_GET['page'];
+                        } else {
+                            $_GET['page'] = 1;
+                        }
+                    }
+                    $products_split = new splitPageResults($_GET['page'], $max_results, $products_query_raw, $products_query_numrows);
+                    $products = $db->Execute($products_query_raw);
+                    // Split Page
+
+                    foreach ($products as $product) {
+                        $products_count++;
+// Get categories_id for product if search
+                        if (isset($_GET['search'])) {
+                            $cPath = $product['master_categories_id'];
+                        }
+
+                        if ((!isset($_GET['pID']) && !isset($_GET['cID']) || (isset($_GET['pID']) && ($_GET['pID'] === $product['products_id']))) && !isset($pInfo) && !isset($cInfo)
```

Comment:
$_GET['pID'] is sanitized to an int, while the database return of the 'products_id' is a string.

Reply:
Prior to correcting this, perhaps this clause needs a review.
I'm not sure what cID/$cInfo has to do with this.

Putting it In a more readable form:
```
if (
                    (
                        !isset($_GET['pID']) 
                        && !isset($_GET['cID']) 
                        || (isset($_GET['pID']) && ($_GET['pID'] === $product['products_id']))
                    )
                    && !isset($pInfo)
                    && !isset($cInfo)
                    && (strpos($action,'new') !== 0))
   {
      $pInfo = new objectInfo($product);
    }
```

Reply:
You have to handle these long lines yourself or you'll hit a hard break.  Something like this:

```
                        if ((!isset($_GET['pID']) && !isset($_GET['cID']) || 
                                (isset($_GET['pID']) && ($_GET['pID'] === $product['products_id']))) 
                            && !isset($pInfo) && !isset($cInfo) && (strpos($action,'new') !== 0)) {
```